aten/src/ATen/native/cuda/Normalization.cu

```diff
@@ -56,8 +73,8 @@ void batch_norm_elementwise(
     AT_DISPATCH_FLOATING_TYPES_AND2(kBFloat16, kHalf, self.scalar_type(),
                                     "batch_norm_elementwise_cuda", [&] {
       using accscalar_t = at::acc_type<scalar_t, true>;
-      if ((weight->defined() && weight->scalar_type() != self.scalar_type()) ||
-          (bias->defined() && bias->scalar_type() != self.scalar_type())) {
+      const bool mixed_type = is_mixed_type(self, *weight, *bias);
```

Comment:
should we also check that bias and weight have same dtype?

Reply:
The `generic_packed_accessor` calls already enforce that they have the same type, so I've just improved the error message when a type doesn't match.